controllers/templateupdaterequest/templateupdaterequest_controller_test.go

```diff
@@ -690,6 +692,159 @@ func TestReconcile(t *testing.T) {
 
 	})
 
+	t.Run("spaces", func(t *testing.T) {
+		t.Run("when space tier hash label still the same", func(t *testing.T) {
+			// given
+			previousBasicTier := tiertest.BasicTier(t, tiertest.PreviousBasicTemplates)
+			basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates, tiertest.WithCurrentUpdateInProgress())
+			space := spacetest.NewSpace("user-1", spacetest.WithTierNameAndHashLabelFor(previousBasicTier)) // space's hash matches TUR hash
+			initObjs := []runtime.Object{basicTier, space}
+			previousHash, err := tierutil.ComputeHashForNSTemplateTier(previousBasicTier)
+			require.NoError(t, err)
+			initObjs = append(initObjs, turtest.NewTemplateUpdateRequest("user-1", *basicTier, turtest.CurrentTierHash(previousHash)))
+			r, req, cl := prepareReconcile(t, initObjs...)
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+			// then
+			require.NoError(t, err)
+			require.Equal(t, reconcile.Result{}, res) // no need to requeue, the Space is watched
+			turtest.AssertThatTemplateUpdateRequest(t, "user-1", cl).
+				HasConditions(toolchainv1alpha1.Condition{
+					Type:   toolchainv1alpha1.TemplateUpdateRequestComplete,
+					Status: corev1.ConditionFalse,
+					Reason: toolchainv1alpha1.TemplateUpdateRequestUpdatingReason,
+				})
+		})
+
+		t.Run("when space tier hash label has changed", func(t *testing.T) {
+			t.Run("space condition not ready yet", func(t *testing.T) {
+				// given
+				previousBasicTier := tiertest.BasicTier(t, tiertest.PreviousBasicTemplates)
+				basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates, tiertest.WithCurrentUpdateInProgress())
+				space := spacetest.NewSpace("user-1", spacetest.WithTierNameAndHashLabelFor(basicTier)) // space's hash updated to new tier
```

Comment:
the condition of the space is actually not set at all - which doesn't have to be clear from the code. It would be nice to set the condition to "updating" as part of the test setup.

Reply:
Done in a569ce2497328bebf5cf79199df106595cc108a1